aws/resource_aws_db_proxy_endpoint.go

```diff
@@ -0,0 +1,229 @@
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateRdsIdentifier,
+			},
+			"db_proxy_endpoint_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateRdsIdentifier,
+			},
+			"endpoint": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"is_default": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"tags": tagsSchema(),
+			"target_role": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      rds.DBProxyEndpointTargetRoleReadWrite,
+				ValidateFunc: validation.StringInSlice(rds.DBProxyEndpointTargetRole_Values(), false),
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vpc_security_group_ids": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"vpc_subnet_ids": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceAwsDbProxyEndpointCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+
+	dbProxyName := d.Get("db_proxy_name").(string)
+	dbProxyEndpointName := d.Get("db_proxy_endpoint_name").(string)
+
+	params := rds.CreateDBProxyEndpointInput{
+		DBProxyName:         aws.String(dbProxyName),
+		DBProxyEndpointName: aws.String(dbProxyEndpointName),
+		TargetRole:          aws.String(d.Get("target_role").(string)),
+		VpcSubnetIds:        expandStringSet(d.Get("vpc_subnet_ids").(*schema.Set)),
+		Tags:                keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().RdsTags(),
+	}
+
+	if v := d.Get("vpc_security_group_ids").(*schema.Set); v.Len() > 0 {
+		params.VpcSecurityGroupIds = expandStringSet(v)
+	}
+
+	_, err := conn.CreateDBProxyEndpoint(&params)
+
+	if err != nil {
+		return fmt.Errorf("error Creating RDS DB Proxy Endpoint (%s/%s): %w", dbProxyName, dbProxyEndpointName, err)
+	}
+
+	d.SetId(strings.Join([]string{dbProxyName, dbProxyEndpointName}, "/"))
+
+	if _, err := waiter.DBProxyEndpointAvailable(conn, d.Id(), d.Timeout(schema.TimeoutCreate)); err != nil {
+		return fmt.Errorf("error waiting for RDS DB Proxy Endpoint (%s) to become available: %w", d.Id(), err)
+	}
+
+	return resourceAwsDbProxyEndpointRead(d, meta)
+}
+
+func resourceAwsDbProxyEndpointRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).rdsconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	dbProxyEndpoint, err := finder.DBProxyEndpoint(conn, d.Id())
+
+	if tfawserr.ErrCodeEquals(err, rds.ErrCodeDBProxyNotFoundFault) {
+		log.Printf("[WARN] RDS DB Proxy Endpoint (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if tfawserr.ErrCodeEquals(err, rds.ErrCodeDBProxyEndpointNotFoundFault) {
```

Comment:
Reference: https://github.com/hashicorp/terraform-provider-aws/issues/16796

```suggestion
	if !d.IsNewResource() && tfawserr.ErrCodeEquals(err, rds.ErrCodeDBProxyEndpointNotFoundFault) {
```